.github/workflows/test.yaml

```
@@ -84,5 +94,48 @@ jobs:
             kubectl rollout status daemonset.apps/image-prefetch --timeout=90s ||
             kubectl describe daemonset.apps/image-prefetch
           }
-      - env: { PGO_TEST_TIMEOUT_SCALE: 1.2 }
-        run: make createnamespaces check-envtest-existing
+      - run: make createnamespaces check-envtest-existing
+        env:
+          PGO_TEST_TIMEOUT_SCALE: 1.2
+          GO_TEST: go test --coverprofile 'envtest-existing.coverage' --coverpkg ./internal/...
+
+      # Upload coverage to GitHub
+      - run: gzip envtest-existing.coverage
+      - uses: actions/upload-artifact@v2
+        with:
+          name: "kubernetes-k3d=${{ matrix.kubernetes }}"
+          path: envtest-existing.coverage.gz
+          retention-days: 1
+
+  coverage-report:
+    runs-on: ubuntu-latest
+    needs:
+      - kubernetes-api
+      - kubernetes-k3d
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with: { go-version: 1.x }
+      - uses: actions/download-artifact@v2
+        with: { path: download }
+
+      # Combine the coverage profiles by taking the mode line from any one file
+      # and the data from all files. Print a list of functions with less than
+      # 100% coverage, and upload a complete HTML report.
+      - name: Generate report
+        run: |
+          gunzip --keep download/*/*.gz
+          ( sed -e '1q' download/*/*.coverage
+            tail -qn +2 download/*/*.coverage ) > total.coverage
+          go tool cover --func total.coverage -o total-coverage.txt
+          go tool cover --html total.coverage -o total-coverage.html
+          sed -e '/100.0%/d' -e "s,$(go list -m)/,," total-coverage.txt
+          echo "::notice title=Total Coverage::$(awk 'END { print $3 }' total-coverage.txt)"
```

🤔 The last line of the text report clarifies what the total percentage is about, `total:	(statements)	89.7%`. We probably should keep that context so the annotation reads: `::notice title=Total Coverage::89.7% (statements)`.

```suggestion
          echo "::notice title=Total Coverage::$(awk 'END { print $3 " " $2 }' total-coverage.txt)"
```